redwood-treehouse-guest/src/commonMain/kotlin/app/cash/redwood/treehouse/StandardAppLifecycle.kt

```diff
@@ -36,8 +36,12 @@ public class StandardAppLifecycle(
   private var started = false
   private lateinit var host: Host
 
-  private lateinit var broadcastFrameClock: BroadcastFrameClock
-  internal lateinit var frameClock: MonotonicFrameClock
+  private val broadcastFrameClock: BroadcastFrameClock = BroadcastFrameClock {
+    if (started) {
```

Comment:
FYI, I confirmed that the host code always calls start() before it does anything else so there’s no hazard of losing a frame here.

The only thing that could do it is the main function that binds the app, which exposes the app lifecycle. Like if you used a composition in your navigation system or something.